metadatastore/test/test_mds.py

```diff
@@ -1,11 +1,401 @@
+        timestamps = {k: ttime.time() for k in data_keys}
+
+        _time = ttime.time()
+        uid = str(uuid.uuid4())
+        all_data.append({'data': data, 'timestamps': timestamps,
+                         'seq_num': seq_num, 'time': _time,
+                         'uid': uid})
+    return all_data
+
+
+@pytest.mark.parametrize('ver', (0, 1))
+def test_pickle(ver):
     md = MDS(config={'host': 'portland'}, version=1)
     md2 = pickle.loads(pickle.dumps(md))
 
     assert md.version == md2.version
     assert md.config == md2.config
+
+
+def test_event_descriptor_insertion(mds_all):
+    mds = mds_all
+    # format some data keys for insertion
+    data_keys = {'some_value': {'source': 'PV:pv1',
+                                'shape': [1, 2],
+                                'dtype': 'array'},
+                 'some_other_val': {'source': 'PV:pv2',
+                                    'shape': [],
+                                    'dtype': 'number'},
+                 'data_key3': {'source': 'PV:pv1',
+                               'shape': [],
+                               'dtype': 'number',
+                               'external': 'FS:foobar'}}
+
+    document_insertion_time = ttime.time()
+    run_start_uid = mds.insert_run_start(scan_id=3022013,
+                                         beamline_id='testbed',
+                                         owner='tester',
+                                         group='awesome-devs',
+                                         project='Nikea',
+                                         time=document_insertion_time,
+                                         uid=str(uuid.uuid4()))
+    time = ttime.time()
+    # test insert
+    ev_desc_uid = mds.insert_descriptor(run_start_uid, data_keys, time,
+                                        str(uuid.uuid4()))
+    ev_desc_mds, = mds.find_descriptors(uid=ev_desc_uid)
+    # make sure the sanitized event descriptor has no uid
+    check_for_id(ev_desc_mds)
+
+    # make sure the event descriptor is pointing to the correct run start
+    referenced_run_start = ev_desc_mds['run_start']
+    assert referenced_run_start.uid == run_start_uid
+    assert ev_desc_mds['time'] == time
+
+    for k in data_keys:
+        for ik in data_keys[k]:
+            assert ev_desc_mds.data_keys[k][ik] == data_keys[k][ik]
+
+
+def test_custom_warn(mds_all):
+    mds = mds_all
+    run_start_uid = str(uuid.uuid4())
+    warnings.simplefilter('always', UserWarning)
+    with pytest.warns(UserWarning):
+        run_start_uid = mds.insert_run_start(
+            scan_id=30220, beamline_id='testbed',
+            owner='Al the Aardvark', group='Orycteropus',
+            project='Nikea', time=ttime.time(),
+            uid=run_start_uid, custom={'order': 'Tubulidentata'})
+
+    rs = next(mds.find_run_starts(order='Tubulidentata'))
+    assert rs['uid'] == run_start_uid
+
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
In this example, the keyword 'custom' is used,  and that is an old feature now deprecated. (It is faster and simpler if custom fields are entered as extra keyword arguments instead of being nested inside a dictionary the way they are here.) When a user passes extra metadata through 'custom' it still works but it raises a warning recommending passing keyword arguments instead.